src/components/ha-date-input.ts

```diff
@@ -1,127 +1,69 @@
-import "@polymer/paper-input/paper-input";
-import type { PaperInputElement } from "@polymer/paper-input/paper-input";
-import {
-  css,
-  customElement,
-  html,
-  LitElement,
-  property,
-  TemplateResult,
-} from "lit-element";
+import "@vaadin/vaadin-date-picker/theme/material/vaadin-date-picker";
 
-@customElement("ha-date-input")
-export class HaDateInput extends LitElement {
-  @property() public year?: string;
+const VaadinDatePicker = customElements.get("vaadin-date-picker");
 
-  @property() public month?: string;
+export class HaDateInput extends VaadinDatePicker {
+  constructor() {
+    super();
 
-  @property() public day?: string;
-
-  @property({ type: Boolean }) public disabled = false;
+    this.i18n.formatDate = this._formatISODate;
+    this.i18n.parseDate = this._parseISODate;
```

Comment:
We should show the date in the format of the user language, we have functions for that: <https://github.com/home-assistant/frontend/blob/dev/src/common/datetime/format_date.ts#L4>

Reply:
Something like `formatDateShort(new Date(d.year, d.month, d.day), this.hass.language)`